docs/Measuring_Resonances.md

```diff
@@ -65,45 +41,116 @@ the system that may damage the electronics.
+script to setup `numpy` and `matplotlib` to get started:
+
 ```
-~/klippy-env/bin/pip install -v numpy
+~/klipper/scripts/setup-shaper-calibrate.sh
 ```
-to install `numpy` package. Note that, depending on the performance of the
+
+Note that, depending on the performance of the
 CPU, it may take *a lot* of time, up to 10-20 minutes. Be patient and wait
 for the completion of the installation. On some occasions, if the board has
-too little RAM, the installation may fail and you will need to enable swap.
+too little RAM, the installation may fail, and you will need to
+[enable swap](https://nebl.io/neblio-university/enabling-increasing-raspberry-pi-swap/).
 
-Next, run the following commands to install the additional dependencies:
-```
-sudo apt update
-sudo apt install python-numpy python-matplotlib
-```
+## Wiring Option 1: Direct Raspberry Pi Connection
+
+The recommended way to connect the sensor to the printer is to connect it directly to
+the SPI bus on the Raspberry Pi.
+
+You need to connect ADXL345 to your Raspberry Pi via SPI. Note that the I2C
+connection, which is suggested by ADXL345 documentation, has too low throughput
+and **will not work**. The recommended connection scheme:
+
+| ADXL345 pin  | RPi pin | RPi pin name |
+| :--:         | :--:    | :--: |
+| 3V3 (or VCC) | 01      | 3.3v DC power |
+| GND          | 06      | Ground |
+| CS           | 24      | GPIO08 (SPI0_CE0_N) |
+| SDO          | 21      | GPIO09 (SPI0_MISO) |
+| SDA          | 19      | GPIO10 (SPI0_MOSI) |
+| SCL          | 23      | GPIO11 (SPI0_SCLK) |
+
+Fritzing wiring diagrams for some of the ADXL345 boards:
+
+![ADXL345-Rpi](img/adxl345-fritzing.png)
 
-Afterwards, check and follow the instructions in the
-[RPi Microcontroller document](RPi_microcontroller.md) to setup the
-"linux mcu" on the Raspberry Pi.
+Double-check your wiring before powering up the Raspberry Pi to prevent
+damaging it or the accelerometer.
 
-Make sure the Linux SPI driver is enabled by running `sudo
-raspi-config` and enabling SPI under the "Interfacing options" menu.
+Check and follow the
+instructions in the [RPi Microcontroller document](RPi_microcontroller.md) to setup the
+"linux mcu" on the Raspberry Pi. Make sure the Linux SPI driver is enabled by running
+`sudo raspi-config` and enabling SPI under the "Interfacing options" menu.
 
 Add the following to the printer.cfg file:
+
 ```
 [mcu rpi]
 serial: /tmp/klipper_host_mcu
 
 [adxl345]
-cs_pin: rpi:None
+cs_pin: rpi: None
 
 [resonance_tester]
 accel_chip: adxl345
 probe_points:
-    100,100,20  # an example
+    100,100,20  # It is advised to start with 1 probe point, in the middle
+    of the print bed, slightly above it.
 ```
-It is advised to start with 1 probe point, in the middle of the print bed,
-slightly above it.
+
+Restart Klipper via the `RESTART` command.
+
+## Wiring Option 2: MCU Connection (Advanced Users)
+If you want to avoid installing the Linux MCU software on the Raspberry Pi, your Pi's
```

Comment:
To be honest, I wouldn't list 'avoid installing the Linux MCU software' as a reason to use MCU connection. Because, frankly, the alternative is a lot more bother.

Reply:
I think which is more of a bother is a matter of opinion that will change depending on who is reading the document. I prefer to run a very light linux installation because installing anything on any computer always introduces more possibility that something could go wrong. Now, is the likelihood that the linux mcu will be cause for an issue with the Klipper system down the line high? Probably not. But installing the linux mcu requires modifying your Klipper start script, and there seems to be no clear way to uninstall the linux mcu so it could also be argued that the ability to simply unplug a secondary mcu makes the linux mcu more of a bother. We've already made it a completely separate option and labeled it for advanced users, I think it's good enough.